scripts/hid_configurator/configurator_core.py

```diff
@@ -620,5 +623,74 @@ def is_dfu_image_correct(dfu_image):
     return True
 
 
+class Step:
+    def __init__(self, r, g, b, substep_count, substep_time):
+        self.r = int(r)
+        self.g = int(g)
+        self.b = int(b)
+        self.substep_count = int(substep_count)
+        self.substep_time = int(substep_time)
+
+
+def led_send_single_step(dev, recipient, step):
+    event_id = (EVENT_GROUP_LED_STREAM << GROUP_FIELD_POS) | (STREAM_DATA << TYPE_FIELD_POS)
+
+    event_data = struct.pack('<BBBHH', step.r, step.g, step.b, step.substep_count, step.substep_time)
+
+    try:
+        success = exchange_feature_report(dev, recipient, event_id, event_data, False)
+    except:
+        success = False
+
+    return success
+
+
+def led_send_random_color_steps(dev, recipient, n):
+    logging.debug('Sending random color steps')
+    substep_count = 10
+    substep_time = 1
+    for i in range(n):
+        logging.debug('Sending {} step'.format(i+1))
+        r = random.randint(0, 255)
+        g = random.randint(0, 255)
+        b = random.randint(0, 255)
+        step = Step(r, g, b, substep_count, substep_time)
+        led_send_single_step(dev, recipient, step)
+
+
+def fetch_free_steps_buffor_info(dev, recipient):
+    event_id = (EVENT_GROUP_LED_STREAM << GROUP_FIELD_POS)
+
+    try:
+        success, fetched_data = exchange_feature_report(dev, recipient, event_id, None, True)
+    except:
+        success = False
+
+    if not success:
+        return None
+
+    fmt = '<B'
+    assert struct.calcsize(fmt) <= EVENT_DATA_LEN_MAX
+
+    if (fetched_data is None) or (len(fetched_data) < struct.calcsize(fmt)):
+        return None
+
+    return struct.unpack(fmt, fetched_data)[0]
+
+def send_continuous_led_stream(dev, recipient):
+    while 1:
+        free_places = fetch_free_steps_buffor_info(dev, recipient)
+        if free_places == None:
+            return None
+        logging.debug('free places: {}'.format(free_places))
+        while free_places < 1:
```

Comment:
Maybe something like:
`if free_places == 0:
    time.sleep(0.1)
else:
    led_send_random_color_steps(...)`

Reply:
btw. how are we supposed to finish this loop ?

Reply:
added signal_handler, done